### DISCUSSION TITLE
No FAN_MODE with S30

### DISCUSSION THREAD

**Discussion by bryanz** [PeteRager/lennoxs30#342] [Q&A] [answered]
Thank you for the S30 integration, it's a huge help! The only problem I've found is that I don't see FAN_MODE, but I do see HVAC_MODE and all the other entities. Do I need to manually configure something? Is there a log I should post for troubleshooting?

**Comment by PeteRager** [comment, marked as answer]
The fan mode is part of the climate entitiy.

https://www.home-assistant.io/integrations/climate/#action-climateset_fan_mode

Valid values are "on" or "auto"

**Reply by bryanz** [reply]
That did it, thank you! I'm new to Home Assistant (but long time tech person), can you explain why it's part of the climate entity and not part of the device like "HVAC_MODE" is?

I have an automation to turn off my heat pump and ventilation, and was using the device rather than the climate entity, is that wrong? Screenshot attached.
![Screenshot 2024-09-20 at 7 55 44 AM](https://github.com/user-attachments/assets/1f4a1b06-dfb4-4be4-bbad-ac6605982491)

**Comment by PeteRager** [comment]
Sounds like the HA device Id implementation is incomplete.  

You'll be better off using entity ids rather than device ids which are opaque uuids.  As automations become more numerous most folks start breaking them up into packages and manage the YAML directly, and then it's much easier to understand what is being done when entity ids are being used.  Myself, I manage all the config in packages and use a github repo for version control.  

https://community.home-assistant.io/t/why-and-how-to-avoid-device-ids-in-automations-and-scripts/605517
